### PR TITLE
chore: Rename `DemoWarnings` to `ProjectNotice` and lemonize it

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

This notice was still Ant-style, and I've been seeing it quite a lot since I've started using `generate_demo_data` for local dev (+ users see the other two variants).

Screen Shot 2022-08-23 at 18 15 48
Screen Shot 2022-08-23 at 18 17 01
Screen Shot 2022-08-23 at 18 16 29

## Changes

Lemonized (or, as some say, lemonified) the notice. The revamp made it low-key, but I think this might be considered an improvement. Also renamed it to `ProjectNotice`, as it was confusingly called `DemoWarnings` (which was perhaps accurate in the beginning, but not for a long time).

Screen Shot 2022-08-23 at 18 17 41
Screen Shot 2022-08-23 at 18 18 29
Screen Shot 2022-08-23 at 18 18 01

### CODE REVIEW

File: frontend/src/lib/components/AlertMessage/AlertMessage.tsx [hunk 1]
```diff
@@ -2,22 +2,27 @@ import React from 'react'
 import './AlertMessage.scss'
 import { IconClose, IconInfo, IconWarning } from '../icons'
 import clsx from 'clsx'
-import { LemonButton } from '../LemonButton'
+import { LemonButton, SideAction } from '../LemonButton'
+import { LemonButtonPropsBase } from '../LemonButton/LemonButton'
+
+export type AlertMessageAction = SideAction & Pick<LemonButtonPropsBase, 'children'>
```
Comment (benjackwhite) [inline frontend/src/lib/components/AlertMessage/AlertMessage.tsx:8]: 🙌

**Review by benjackwhite — APPROVED** [review]
Nice! I thought the same after looking at the demo. I feel low key style is better tbh.

I wonder if we can call this AlertBanner feature complete now it has the nice action prop https://github.com/orgs/PostHog/projects/55/views/1

**Review by clarkus — COMMENTED** [review]
The component changes look good to me - I do question the placement, though. If we're describing a page state, placing the alert above the breadcrumbs and header feels misaligned to the page hierarchy. Not a huge concern, but it does displace all the components that give the user a sense of place and orientation within the product. 

If the intent is to represent a global (to the project, or to the organization) constraint, then the [banner position above the main app header](https://www.figma.com/file/Y9G24U4r04nEjIDGIEGuKI/PostHog-Design-System-One?node-id=3706%3A1999) might be a better placement?

**Comment by clarkus** [comment]
Worth mentioning - some of these banner scenarios would also work well as [timely in-app notifications](https://www.figma.com/file/Y9G24U4r04nEjIDGIEGuKI/PostHog-Design-System-One?node-id=3112%3A1296) (if/when we get around to building that feature).

**Comment by Twixes** [comment]
I agree the placement is not ideal @clarkus. It's quite annoying technically to place things under the header though, because that's implemented individually by each page and not on the nav level component (like breadcrumbs or this banner). In the future I think it'd be great to standardize the header system – then we could be more flexible with the placement of info like this.

**Pull request merged**